projects/ichifarm/index.js

```diff
@@ -0,0 +1,221 @@
+          abi: abiFarm.getCurrentTokens,
+          target: tokenLp,
+          block,
+        })
+      ).output;
+
+      for (const token of getCrTokens) {
+        await sumTokensAndLPsSharedOwners(
+          balances,
+          [[token, false]],
+          [tokenLp]
+        );
+      }
+    } else {
+      sdk.util.sumSingleBalance(balances, tokenLp, balanceLp);
+    }
+  }
+
+  await unwrapUniswapLPs(balances, lpPositions, block);
+};
+
+const ethTvl = async (block) => {
+  const balances = {};
+
+  /*** ICHI oneTokens V1 TVL Portion ***/
+  const assetsList = (await fetchURL(endpoint)).data.tokens
+    .map((addr) => addr.address)
+    .concat(USDC);
+
+  for (const asset of assetsList) {
+    await sumTokensAndLPsSharedOwners(balances, [[asset, false]], oneTokensV1);
+  }
+
+  /*** ICHI Farm V1 TVL Portion ***/
+  await calc(
+    block,
+    balances,
+    farmV1Contract,
+    abiFarm.poolLength,
+    abiFarm.getPoolToken,
+    abiFarm.getLPSupply
+  );
+
+  /*** ICHI oneTokens V2 TVL Portion ***/
+  const countOneTokens = (
+    await sdk.api.abi.call({
+      abi: abiOneToken.oneTokenCount,
+      target: OneTokenFactoryContract,
+      block,
+    })
+  ).output;
+
+  for (let i = 0; i < countOneTokens; i++) {
+    const oneToken = (
+      await sdk.api.abi.call({
+        abi: abiOneToken.oneTokenAtIndex,
+        target: OneTokenFactoryContract,
+        params: i,
+        block,
+      })
+    ).output;
+
+    const countAssets = (
+      await sdk.api.abi.call({
+        abi: abiOneToken.assetCount,
+        target: oneToken,
+        block,
+      })
+    ).output;
+
+    for (let i = 0; i < countAssets; i++) {
+      const asset = (
+        await sdk.api.abi.call({
+          abi: abiOneToken.assetAtIndex,
+          target: oneToken,
+          params: i,
+          block,
+        })
+      ).output;
+
+      const assetBalInVault = Number(
+        (
+          await sdk.api.abi.call({
+            abi: abiOneToken.balances,
+            target: oneToken,
+            params: asset,
+            block,
+          })
+        ).output.inVault
+      );
+
+      const assetBalInStrategy = Number(
+        (
+          await sdk.api.abi.call({
+            abi: abiOneToken.balances,
+            target: oneToken,
+            params: asset,
+            block,
+          })
+        ).output.inStrategy
+      );
+
+      const totalBalance = (
+        assetBalInVault + assetBalInStrategy
+      ).toLocaleString("fullwide", {
+        useGrouping: false,
+      });
+
+      sdk.util.sumSingleBalance(balances, asset, totalBalance);
+    }
+  }
+
+  /*** ICHI Farm V2 TVL Portion ***/
+  await calc(
+    block,
+    balances,
+    farmV2Contract,
+    abiFarm.poolLength,
+    abiFarm.lpToken,
+    abiFarm.getLPSupply
+  );
+  return balances;
+};
+
+module.exports = {
+  misrepresentedTokens: true,
+  ethereum: {
+    staking: staking(stakingContract, ICHI),
+    stakingFarmV1: staking(farmV1Contract, ICHI),
```

Comment:
keys other than the standard ones (eg: staking, pool2, tvl...) will be ignored by our frontend so may be better to just count it all in a single function

Reply:
I take note. Do you want me modify it?

Reply:
yeah would be great

Reply:
Done